### ISSUE TITLE
__webpack_public_path__ does not work

### ISSUE THREAD

**Issue by yiminghe** [webpack/webpack#842]
according to http://webpack.github.io/docs/configuration.html#output-publicpath
```
If you don’t know the publicPath while compiling 
you can omit it and set __webpack_public_path__ on runtime.
```

but runtime generated by 1.6.0 does not consider `__webpack_public_path__` ,for example
```js
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = "";
``` 

it should be
```js
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = window.__webpack_public_path__ || "";
```

**Comment by sokra** [comment]
Just use `__webpack_public_path__` in your bundle. It's only valid in your bundle.

**Comment by yiminghe** [comment]
Thanks, understood, webpack will compile 
```js
__webpack_public_path__ = '/build/example/';
```

in commonjs source file into

```js
__webpack_require__.p = '/build/example/';
```

Docs need to be more clear :)

**Comment by molszanski** [comment]
A friendly note to CoffeeScript developers

```coffee
__webpack_public_path__ = "something"
```

Will compile to:

```js
var __webpack_public_path__; 
__webpack_public_path__ = "something"
```

that will not work.

Use something like this:
```coffee
`__webpack_public_path__ = "something"`
```
that will embed a pure js

**Comment by james-erekson-sportsbet** [comment]
A note for TypeScript developers, TypeScript won't let us set variables on the window when **noImplicitAny** is on. So we tried a workaround that normally works for variables we want to attach to window:
```
(<any>window).__webpack_public_path__ = "something"
```
However, the code compiles to:
```
window.__webpack_public_path__ = "something";
```
Which, similar to the CoffeeScript example above, is not replaced by webpack. The only way we were able to get around this was by breaking out of TypeScript and including the file as pure js i.e.
```
__webpack_public_path__ = "something"
```
